neverbleed.c

```diff
@@ -324,10 +343,33 @@ static int expbuf_write(struct expbuf_t *buf, int fd)
     return 0;
 }
 
+static void yield_on_data(int fd)
+{
+    fd_set rfds;
+    int ret;
+    FD_ZERO(&rfds);
+    FD_SET(fd, &rfds);
+
+    while((ret = select(fd + 1, &rfds, NULL, NULL, NULL)) == -1 && (errno == EAGAIN || errno == EINTR))
+        ;
+    if (ret == -1) {
+        fprintf(stderr, "error in select(2): %d, %s\n", errno, strerror(errno));
+        dief("select(2)");
+    } else if (ret > 0) {
+        // yield when data is available
+        struct timespec tv = { .tv_nsec = 1 };
+        if (-1 == nanosleep(&tv, NULL)) {
+            dief("nanosleep");
```

Comment:
`nanosleep` returns -1 when it is interrupted by a signal. I do not think we want to die when receiving an arbitrary signal?